### PR TITLE
Cleanup an inactive directive in serialization

### LINKED ISSUE #3061: Clean up unused directives in serialization code

### PR DESCRIPTION
**Pull request by khdang**
Clean up MERGE_DCJS directive for data contract serializers.
Fix #3061
cc: @SGuyGe @shmao

### CODE REVIEW

File: src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/XmlObjectSerializerWriteContextComplexJson.cs [hunk 1]
```diff
@@ -154,16 +126,12 @@ protected override bool WriteTypeInfo(XmlWriterDelegator writer, DataContract co
                  (contract.Name.Value == declaredContract.Name.Value &&
                  contract.Namespace.Value == declaredContract.Namespace.Value)) &&
                  (contract.UnderlyingType != Globals.TypeOfObjectArray)
-#if NET_NATIVE || MERGE_DCJS
-                && (_emitXsiType != EmitTypeInformation.Never)
-#endif
-                )
+                && (_emitXsiType != EmitTypeInformation.Never))
```
Comment (stephentoub) [inline src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/XmlObjectSerializerWriteContextComplexJson.cs]: Nit: the && can move to the previous line to be consistent with the others

**Comment by stephentoub** [comment]
LGTM

File: src/System.Private.DataContractSerialization/src/System.Private.DataContractSerialization.csproj [hunk 2]
```diff
@@ -48,7 +47,7 @@
     <Compile Include="$(RuntimeSerializationSources)\ExtensionDataObject.cs" Condition="$(NetNative)" />
     <Compile Include="$(RuntimeSerializationSources)\ExtensionDataReader.cs" Condition="$(NetNative)" />
     <Compile Include="$(RuntimeSerializationSources)\Globals.cs" />
-    <Compile Include="$(RuntimeSerializationSources)\GenericParameterDataContract.cs" Condition="$(NetNative) Or $(MergeDcjs)" />
```
Comment (SGuyGe) [inline src/System.Private.DataContractSerialization/src/System.Private.DataContractSerialization.csproj]: Why remove $(NetNative)? NetNative is still there.

File: src/System.Private.DataContractSerialization/src/System.Private.DataContractSerialization.csproj [hunk 3]
```diff
@@ -73,9 +72,9 @@
     <Compile Include="$(RuntimeSerializationSources)\TypeCode.cs" />
     <Compile Include="$(RuntimeSerializationSources)\ExceptionDataContract.cs" />
     <Compile Include="$(RuntimeSerializationSources)\KnownTypeDataContractResolver.cs" />
-    <Compile Include="$(RuntimeSerializationSources)\XmlObjectSerializerReadContextComplex.cs" Condition="$(NetNative) Or $(MergeDcjs)" />
-    <Compile Include="$(RuntimeSerializationSources)\XmlObjectSerializerWriteContextComplex.cs" Condition="$(NetNative) Or $(MergeDcjs)" />
-    <Compile Include="$(RuntimeSerializationSources)\BitFlagsGenerator.cs" Condition="!$(NetNative) And $(MergeDcjs)" />
```
Comment (SGuyGe) [inline src/System.Private.DataContractSerialization/src/System.Private.DataContractSerialization.csproj]: Ditto.

**Comment by shmao** [comment]
LGTM.

File: src/System.Private.DataContractSerialization/src/System.Private.DataContractSerialization.csproj (hunk 2 above)
Comment (khdang) [inline src/System.Private.DataContractSerialization/src/System.Private.DataContractSerialization.csproj]: We're using MERGE_DCJS directive so the $MergeDcjs is true. Therefore this condition is always true $(NetNative) Or $(MergeDcjs) so I've removed the condition altogether.

File: src/System.Private.DataContractSerialization/src/System.Private.DataContractSerialization.csproj (hunk 3 above)
Comment (khdang) [inline src/System.Private.DataContractSerialization/src/System.Private.DataContractSerialization.csproj]: As previous comment.

**Comment by SGuyGe** [comment]
LGTM

**Comment by khdang** [comment]
@dotnet-bot test this please

**Comment by khdang** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
Could you please squash this, in particular to get rid of the merge?

**Comment by khdang** [comment]
I've squashed the commits into one.

**Pull request merged**